### PR TITLE
Add UART logger

### PR DESCRIPTION
**Pull request by stephen99scott**
* Rebased hs/logger_Uart to master
* Added early init of log
* Made changes for nucleo support
* Abstract clock source

### CODE REVIEW

File: firmware/applications/SignBuddy/main.c [hunk 1]
```diff
@@ -50,6 +54,7 @@ void sysclk_init(void)
 
   LL_SetSystemCoreClock(SYSCLK_FREQ);
   LL_RCC_SetLPUARTClockSource(LL_RCC_LPUART1_CLKSOURCE_HSI);
```
Comment (stephen99scott) [inline firmware/applications/SignBuddy/main.c:56]: Will make the same abstraction for the Bluetooth UART on another branch

File: firmware/applications/SignBuddy/nucleo_board.h [hunk 2]
```diff
@@ -34,16 +34,17 @@
...
-#define LOG_UART_RX_PIN           LL_GPIO_PIN_8
-#define LOG_UART_TX_PIN           LL_GPIO_PIN_9
-#define LOG_UART_GPIO_PORT        GPIOD
+#define LOG_UART                  USART2
+#define LOG_UART_IRQn             USART2_IRQn
+#define LOG_UART_IRQHandler       USART2_IRQHandler
+#define LOG_UART_RX_PIN           LL_GPIO_PIN_3
+#define LOG_UART_TX_PIN           LL_GPIO_PIN_2
+#define LOG_UART_GPIO_PORT        GPIOA
 #define LOG_UART_GPIO_AF          LL_GPIO_AF_7
```
Comment (adambujak) [inline firmware/applications/SignBuddy/nucleo_board.h:43]: why did you change these pins?

**Comment by adambujak** [comment]
looks good. did you test logging on the nucleo?

should we delete the rx stuff from the log_uart? I don't think we'll ever be receiving anything and it'll make the code simpler and we'll save 512 bytes

**Comment by stephen99scott** [comment]
> looks good. did you test logging on the nucleo?

Yup, tested and it works as expected
> should we delete the rx stuff from the log_uart? I don't think we'll ever be receiving anything and it'll make the code simpler and we'll save 512 bytes

The only time we'd be receiving anything is if the android app requests a data collection, but it would probably make sense to handle that request separate from logging, so I think we can probably get rid of it. In general, what would be a use case when you would want to receive in the log?

**Comment by adambujak** [comment]
> The only time we'd be receiving anything is if the android app requests a data collection, but it would probably make sense to handle that request separate from logging, so I think we can probably get rid of it. In general, what would be a use case when you would want to receive in the log?

something like changing the logger level, pausing output, etc.

**Review by adambujak — APPROVED** [review]

**Comment by stephen99scott** [comment]
> > The only time we'd be receiving anything is if the android app requests a data collection, but it would probably make sense to handle that request separate from logging, so I think we can probably get rid of it. In general, what would be a use case when you would want to receive in the log?
> 
> something like changing the logger level, pausing output, etc.

Oh okay that makes sense, thanks
